index.js

```diff
@@ -186,39 +127,51 @@ function makeCachedResolver (prefix, fieldName, cache, originalFieldResolver, po
       }
 
       // We must skip ctx and info as they are not easy to serialize
+      // TODO use a fast JSON stringify
+      // TODO can we remove self from the key?
```

Comment:
you need _some_ part of `self` to identify which object are we referring to, e.g. User 42. We could automatically detect if the object as an `ID` field and use that instead. The same logic could be applied to automate some part of the invalidations further.